### ISSUE TITLE
[i18n + redirects] unnecessary redirect for index page

### ISSUE THREAD

**Issue by dmitrykozyrenko** [vercel/next.js#19405]
# Bug report

## Describe the bug

Unnecessary redirect for index page.

i18n config: `{ locales: ['en', 'fr', 'zh'], defaultLocale: 'en' }`
redirects config: `{ source: '/first-page', destination: '/', permanent: true }`

In this case we have three redirects instead of two: `/first-page` -> `/en/` -> `/en`, `/fr/first-page` -> `/fr/` -> `/fr`

## To Reproduce

Described in the `Describe the bug` section.

## Expected behavior

For index page we should redirect only once, from `/first-page` -> `/en`

## System information

- OS: Windows
- Browser: Chrome
- Version of Next.js: 10.0.3-canary.0
- Version of Node.js: 12.16.1
- Deployment: dev, next start, Vercel

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
